### ISSUE TITLE
Tilt do not delete builded image after push

### ISSUE THREAD

**Issue by albaquarto** [tilt-dev/tilt#6196]
I use Tilt with minikube and a local registry. 
Tilt build the images inside my host (docker deamon does not run inside minikube, the CRI is cri-o), push the image to the local registry (is  a  docker container inside my host) and deploy the manifests in minikube that runs as container in my host, inside minikube the CRI, I use cri-o, pulls the image from the local registry and runs the pods.
At the end I have the same image duplicated 3 times in my host: in the docker storage, in the local registry storage and in cri-o storage.

I tried to use tilt docker_prune with these settings disable=False, keep_recent=0, num_builds=1 but images are never deleted from local docker. I wish tilt would deleted the images from docker after it pushes that in the registry or at least at the stop.

Tilt version 0.33.4
Docker version 24.0.5

**Comment by nicks** [comment]
thanks for the request.

hmmm...the tilt garbage collector tries to never delete an image currently in use. That way, if you quit tilt and restart it, everything is cached and reconnects. i don't see us adding a feature like this anytime soon, because it breaks that promise.

There are other issues tracking deleting images from a registry (https://github.com/tilt-dev/tilt/issues/2102), and removing the need for a separate registry (but this is a longer pole that needs deep integration with the container runtime)

A good workaround if you want this is to add a `local_resource` to your tiltfile that prunes the images the way you want.

**Comment by albaquarto** [comment]
> thanks for the request.
> 
> hmmm...the tilt garbage collector tries to never delete an image currently in use. That way, if you quit tilt and restart it, everything is cached and reconnects. i don't see us adding a feature like this anytime soon, because it breaks that promise.
> 
> There are other issues tracking deleting images from a registry (#2102), and removing the need for a separate registry (but this is a longer pole that needs deep integration with the container runtime)
>
Thank you for the answere
I understand your point but the argument "keep_recent" in docker_prune_settings should be used to let the developer decide 
how many images to keep and I decided for 0. Why do not respect this decision? Is a custom config  with a different default. 
There is also a bug because if I set keep_recent=1 with my environment configuration all images also the unused are preserved in my docker. If I use docker as runtime in my minikube the build is done in the cluster and only the last image is preserved. Maybe I have to insert this as issue. 

> A good workaround if you want this is to add a `local_resource` to your tiltfile that prunes the images the way you want.

I do not think that this is a good workaround or maybe I don't understand how to use it because the builds are made async but "Tilt performs an update on local resources on tilt up and whenever any of their deps change." So my local resources is performed before the first build, and I tried to configure a local_resource with resource_deps but if I trigger  the update of the resource_dep the local resource is not triggered. So I don't know how to use a local resource to delete the images.
